src/model/model.ts

```diff
@@ -110,8 +111,13 @@ export abstract class AbstractModel<
     context?: CustomCtx
   ): Promise<MResponse> {
     const start = Date.now();
+
     const mergedContext = deepMerge(this.context, context);
-    const mergedParams = deepMerge(this.params, params);
+    const mergedParams = {
+      ...this.params,
+      ...params,
```

Comment:
did you confirm there is no valid case where params is a nested object? it might even be worth restricting the type if it's easy because that could be a nasty bug.

Reply:
reverted to just using the signal